keps/sig-api-machinery/1929-built-in-default/README.md

```diff
@@ -414,48 +416,64 @@ Both these cases will be handled properly, here's an example:
 type Object struct {
      // This field has an invalid zero-value, empty string is not a
      // valid string.
-     // +default=default-name
-     Name string
+     // +default="default-name"
+     Name string `json:"name,omitempty"`
 
-     // This field is defaulted to 0 with no defaulter.
+     // This field would be defaulted to 0 without a default.
      // +default=0
      Defaulted int
 }
 ```
 
+Non-pointer scalar fields that have a default must be omitempty (unless
+their default is the zero-value). Kubebuilder and built-in defaulting
+generators should both require this.
+
 This would generate the following OpenAPI:
```

Comment:
where is the golang source for the schema below?

Reply:
Ah yes, the name doesn't match, good catch